src/awkward/_v2/_connect/rdataframe/from_rdataframe.py

```diff
@@ -157,7 +157,16 @@ def form_dtype(form):
 
         # Triggers event loop and execution of all actions booked in the associated RLoopManager.
         cpp_reference = result_ptrs.GetValue()
+        lookup = cpp_reference[0].lookup()
+        generator = cpp_reference[0].generator()
+        array_out = generator[column].tolayout(lookup[column], 0, ())
 
-        return _wrap_as_array(column, ak._v2.from_iter(cpp_reference))
+        return ak._v2._util.wrap(
+            ak._v2.contents.RecordArray(
+                fields=[column],
+                contents=[array_out],
+            ),
+            highlevel=True,
+        )
```

Comment:
Isn't this the same as `_maybe_wrap` (which is no longer "maybe")?

And `_wrap_as_array` doesn't seem to be used anymore (i.e. dead code).

Always returning a RecordArray of one field is a step toward returning a collection of columns as a RecordArray of many fields.

Reply:
done